osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+# Set up django configuration
+import django
+from kobo.client.constants import FINISHED_STATES
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+django.setup()
+from kobo.hub.models import Task  # noqa
+
+from osh.hub.scan.models import RetentionPolicySetting  # noqa
+from osh.hub.scan.models import TaskResultsRemoval  # noqa
+
+logger = logging.getLogger("osh.hub.scripts.osh-retention")
+
+
+def delete_tasks_results(tasks_to_delete):
+    for task, retention_policy_setting in tasks_to_delete:
+        try:
+            logger.info('The following task results will be deleted: %s.', task.id)
+            path_to_delete = Task.get_task_dir(task.id)
+            shutil.rmtree(path_to_delete)
+            logging.info('The following folder was deleted:  %s', path_to_delete)
+        except OSError as e:
+            logger.error("Error deleting tasks folder: %s", e)
+            return
+
+        # If the tasks results are applied, we update the database field, so we don't try to delete them again.
+        TaskResultsRemoval.objects.create(task=task, reason=retention_policy_setting)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identify_personal_task(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner
+
+
+def main():
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    # Retrieval of all tasks ids that already have the retention policy applied
+    processed_tasks = TaskResultsRemoval.objects.values_list('task', flat=True)
+    # Retrieval of tasks that has not been applied to retention policy to be applied
+    tasks = Task.objects.filter(state__in=FINISHED_STATES).exclude(id__in=processed_tasks)
+
+    # ---------- Section to take parameters from DB
+    retention_personal_tasks = RetentionPolicySetting.objects.get(category="PersonalScan")
+    retention_failed_status = RetentionPolicySetting.objects.get(category="FailedStatus")
+    # ---------- End Section to take parameters from DB
+
+    eligible_tasks_to_be_deleted = []
+
+    # We identify which tasks are eligible to be deleted
+    today = date.today()
+    for t in tasks:
+        # At the moment, we only have 2 rules for retention settings:
+        # Personal tasks
+        # Failed status
+        calculated_date_personal_tasks = (t.dt_finished + datetime.timedelta(days=retention_personal_tasks.days)).date()
```

Comment:
When I tried to run the script on `covscan.lab.eng.brq2.redhat.com`, it failed with the following traceback before it deleted any results:
```py
# /usr/sbin/osh-retention
Traceback (most recent call last):
  File "/usr/sbin/osh-retention", line 89, in <module>
    main()
  File "/usr/sbin/osh-retention", line 73, in main
    calculated_date_personal_tasks = (t.dt_finished + datetime.timedelta(days=retention_personal_tasks.days)).date()
TypeError: unsupported operand type(s) for +: 'NoneType' and 'datetime.timedelta'
```